client/db/types.go

```diff
@@ -318,19 +322,25 @@ func decodeMatchProof_v0(pushes [][]byte) (*MatchProof, error) {
 }
 
 func decodeMatchProof_v1(pushes [][]byte) (*MatchProof, error) {
-	if len(pushes) != 21 {
+	pushes = append(pushes, nil)
+	return decodeMatchProof_v2(pushes)
+}
+
+func decodeMatchProof_v2(pushes [][]byte) (*MatchProof, error) {
+	if len(pushes) != 22 {
 		return nil, fmt.Errorf("DecodeMatchProof: expected 21 pushes, got %d", len(pushes))
 	}
 	return &MatchProof{
-		Script:        pushes[0],
-		CounterScript: pushes[1],
-		SecretHash:    pushes[2],
-		Secret:        pushes[3],
-		MakerSwap:     pushes[4],
-		MakerRedeem:   pushes[5],
-		TakerSwap:     pushes[6],
-		TakerRedeem:   pushes[7],
-		RefundCoin:    pushes[8],
+		Script:          pushes[0],
+		CounterContract: pushes[1],
+		CounterTxData:   pushes[21],
```

Comment:
Why not `pushes[2]` to maintain the order and aid readability.
In `decodeMatchProof_v1`, instead of appending the nil `[]byte`, can insert:
```
pushes = append(pushes, nil)
copy(pushes[3:], pushes[2:])
pushes[2] = nil
```

Reply:
I don't really see how this would improve readability. Simpler is better in these upgrades, and changing the index of every read of `pushes` for index > 1 is more error-prone.

Reply:
This works for sure, but the readability improvements I mean is having the order of the fields match the order of data in the pushes. So last field in the struct should write to / read from the last data in `pushes`. It'd become a bit of a eye turner if for example, the first struct field is populated with the 10th data in `pushes`, the second field is populated with the data at index 3 and the last field is populated with index 1.

What I'm recommending for clarity is:
```
		Script:          pushes[0],
		CounterContract: pushes[1],
		CounterTxData:   pushes[2],
		SecretHash:      pushes[3],
		Secret:          pushes[4],
		MakerSwap:       pushes[5],
```
instead of
```
		Script:          pushes[0],
		CounterContract: pushes[1],
		CounterTxData:   pushes[21],
		SecretHash:      pushes[2],
		Secret:          pushes[3],
		MakerSwap:       pushes[4],
```